py-polars/tests/unit/io/test_spreadsheet.py

```diff
@@ -194,6 +194,36 @@ def test_read_excel_basic_datatypes(
         assert_frame_equal(df, df)
 
 
+@pytest.mark.parametrize(
+    ("read_spreadsheet", "source", "params"),
+    [
+        (pl.read_excel, "path_xlsx", {"engine": "xlsx2csv"}),
+        (pl.read_excel, "path_xlsx", {"engine": "openpyxl"}),
+        (pl.read_excel, "path_xlsb", {"engine": "pyxlsb"}),
+        (pl.read_ods, "path_ods", {}),
+    ],
+)
+def test_read_invalid_worksheet(
+    read_spreadsheet: Callable[..., dict[str, pl.DataFrame]],
+    source: str,
+    params: dict[str, str],
+    request: pytest.FixtureRequest,
+) -> None:
+    spreadsheet_path = request.getfixturevalue(source)
+    for param, sheet_id, sheet_name in (
+        ("id", 999, None),
+        ("name", None, "not_a_sheet_name"),
+    ):
+        value = sheet_id if param == "id" else sheet_name
+        with pytest.raises(
+            ValueError,
+            match=f"no matching sheets found when `sheet_{param}` is {value!r}",
+        ):
```

Comment:
Are you really still testing when you're just copying the source code into your tests? I'd just leave it at `match="no matching sheets found"`. That's just my preference though.

Reply:
It's just to make sure that the correct param name/value is reported in the error; if anyone modifies it in the future or it accidentally regresses, we'd notice (since in the test I set the expected param/value explicitly, whereas in the code it's inferred)  ;)